dotCMS/src/main/java/com/dotcms/storage/ContentletMetadataAPIImpl.java

```diff
@@ -0,0 +1,301 @@
+                                      final Set<String> fullBinaryFieldNameSet,
+                                      final ImmutableMap.Builder<String, Map<String, Object>> fullMetadataMap,
+                                      final Map<String, Field> fieldMap,
+                                      final boolean alwaysRegenerateMetadata) throws IOException {
+
+        final String storageType        = Config.getStringProperty(DEFAULT_STORAGE_TYPE, StorageType.FILE_SYSTEM.name());
+        Map<String, Object> metadataMap = Collections.emptyMap();
+        final String metadataBucketName = Config.getStringProperty(METADATA_GROUP_NAME, "dotmetadata");
+        for (final String fullBinaryFieldName : fullBinaryFieldNameSet) {
+
+            final File file           = contentlet.getBinary(fullBinaryFieldName);
+            final String metadataPath = this.getFileName(contentlet, fullBinaryFieldName);
+            if (null != file && file.exists() && file.canRead()) {
+
+                final Set<String> metadataFields = this.getMetadataFields(fieldMap.get(fullBinaryFieldName).id());
+                metadataMap = this.fileStorageAPI.generateMetaData(file,
+                        new GenerateMetaDataConfiguration.Builder()
+                            .full(true)
+                            .override(alwaysRegenerateMetadata)
+                            .cache(false)  // do not want cache on full meta
+                            .store(true)
+                            .metaDataKeyFilter(metadataKey -> metadataFields.isEmpty()? true: metadataFields.contains(metadataKey))
+                            .storageKey(new StorageKey.Builder().group(metadataBucketName).key(metadataPath).storage(storageType).build())
+                            .build()
+                        );
+
+                fullMetadataMap.put(fullBinaryFieldName, metadataMap);
+            } else {
+
+                throw new IOException("The file: " + file + ", is null, does not exists or can not access");
+            }
+        }
+    }
+
+    private Set<String> getMetadataFields (final String fieldIdentifier) {
+
+        final Optional<FieldVariable> customIndexMetaDataFieldsOpt =
+                Try.of(()->FactoryLocator.getFieldFactory().byFieldVariableKey(fieldIdentifier, BinaryField.INDEX_METADATA_FIELDS)).getOrElse(Optional.empty());
+
+        final Set<java.lang.String> metadataFields = customIndexMetaDataFieldsOpt.isPresent()?
+                new HashSet<>(Arrays.asList(customIndexMetaDataFieldsOpt.get().value().split(StringPool.COMMA))):
+                this.getConfiguredMetadataFields();
+
+        return metadataFields;
+    }
+
+    /**
+     * Reads INDEX_METADATA_FIELDS from configuration
+     * @return
+     */
+    private Set<java.lang.String> getConfiguredMetadataFields(){
+
+        final java.lang.String configFields = Config.getStringProperty("INDEX_METADATA_FIELDS", null);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unnecessary use of fully qualified name 'java.lang.String' due to existing implicit import 'java.lang.*'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5547873)